cmd/commands/runtime.go

```diff
@@ -458,49 +469,69 @@ func ensureIngressClass(ctx context.Context, opts *RuntimeInstallOptions) error
+	ingressClassNameToController := make(map[string]ingressController)
 	var isValidClass bool
+
 	for _, ic := range ingressClassList.Items {
-		if ic.ObjectMeta.Labels["app.kubernetes.io/name"] == "ingress-nginx" {
-			ingressClassNames = append(ingressClassNames, ic.Name)
-			ingressClassNameToController[ic.Name] = fmt.Sprintf("%s-controller", getControllerName(ic.Spec.Controller))
-			if opts.IngressClass == ic.Name {
-				isValidClass = true
+		for _, controller := range supportedControllers {
+			if ic.Spec.Controller == string(controller) {
+				ingressClassNames = append(ingressClassNames, ic.Name)
+				ingressClassNameToController[ic.Name] = ingressController{
+					Name: getIngressControllerName(controller, ic.Name),
+					Type: controller,
+				}
+
+				if opts.IngressClass == ic.Name { //if ingress class provided via flag
+					isValidClass = true
+				}
+				break
 			}
 		}
 	}
 
-	if opts.IngressClass != "" { //if user provided ingress class by flag
-		if isValidClass {
-			opts.IngressController = ingressClassNameToController[opts.IngressClass]
-			return nil
+	if opts.IngressClass != "" { //if ingress class provided via flag
+		if !isValidClass {
+			return fmt.Errorf("ingress class '%s' is not supported", opts.IngressClass)
 		}
-		return fmt.Errorf("ingress class \"%s\" is not supported. only the ingress class of type nginx is supported.", opts.IngressClass)
 	}
 
 	if len(ingressClassNames) == 0 {
-		return fmt.Errorf("no ingress classes of type nginx were found. please install a nginx ingress controller on your cluster before installing a runtime.")
+		return fmt.Errorf("no ingress classes of the supported types were found")
 	}
-
+	
 	if len(ingressClassNames) == 1 {
 		log.G(ctx).Info("Using ingress class: ", ingressClassNames[0])
 		opts.IngressClass = ingressClassNames[0]
-		opts.IngressController = ingressClassNameToController[opts.IngressClass]
-		return nil
 	}
-
-	if !store.Get().Silent {
-		err = getIngressClassFromUserSelect(ctx, ingressClassNames, &opts.IngressClass)
-		if err != nil {
-			return err
+	
+	if len(ingressClassNames) > 1 {
+		if !store.Get().Silent {
+			err = getIngressClassFromUserSelect(ctx, ingressClassNames, &opts.IngressClass)
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("there are multiple ingress controllers on your cluster, please add the --ingress-class flag and define its value")
 		}
-
-		opts.IngressController = ingressClassNameToController[opts.IngressClass]
-		return nil
 	}
 
-	return fmt.Errorf("please add the --ingress-class flag and define its value")
+	opts.IngressController = ingressClassNameToController[opts.IngressClass].Name
+	opts.IngressControllerType = ingressClassNameToController[opts.IngressClass].Type
+
+	return nil
+}
+
+func getIngressControllerName(controllerType ingressControllerType, className string) string {
+	switch controllerType {
+	case NginxCommunity:
+		return "ingress-nginx-controller"
```

Comment:
this name doesn't sound right. I'm pretty sure it should be `nginx-ingress-controller`